Source/options.cpp

```diff
@@ -712,13 +652,147 @@ std::vector<OptionEntryBase *> ChatOptions::GetEntries()
+{
+}
+void OptionEntryLanguageCode::LoadFromIni(string_view category)
+{
+#ifdef __ANDROID__
+	JNIEnv *env = (JNIEnv *)SDL_AndroidGetJNIEnv();
+	jobject activity = (jobject)SDL_AndroidGetActivity();
+	jclass clazz(env->GetObjectClass(activity));
+	jmethodID method_id = env->GetMethodID(clazz, "getLocale", "()Ljava/lang/String;");
+	jstring jLocale = (jstring)env->CallObjectMethod(activity, method_id);
+	const char *cLocale = env->GetStringUTFChars(jLocale, nullptr);
+	std::string locale = cLocale;
+	env->ReleaseStringUTFChars(jLocale, cLocale);
+	env->DeleteLocalRef(jLocale);
+	env->DeleteLocalRef(activity);
+	env->DeleteLocalRef(clazz);
+#elif defined(__vita__)
+	int32_t language = SCE_SYSTEM_PARAM_LANG_ENGLISH_US; // default to english
+	const char *vita_locales[] = {
+		"ja_JP",
+		"en_US",
+		"fr_FR",
+		"es_ES",
+		"de_DE",
+		"it_IT",
+		"nl_NL",
+		"pt_PT",
+		"ru_RU",
+		"ko_KR",
+		"zh_TW",
+		"zh_CN",
+		"fi_FI",
+		"sv_SE",
+		"da_DK",
+		"no_NO",
+		"pl_PL",
+		"pt_BR",
+		"en_GB",
+		"tr_TR",
+	};
+	SceAppUtilInitParam initParam;
+	SceAppUtilBootParam bootParam;
+	memset(&initParam, 0, sizeof(SceAppUtilInitParam));
+	memset(&bootParam, 0, sizeof(SceAppUtilBootParam));
+	sceAppUtilInit(&initParam, &bootParam);
+	sceAppUtilSystemParamGetInt(SCE_SYSTEM_PARAM_ID_LANG, &language);
+	if (language < 0 || language > SCE_SYSTEM_PARAM_LANG_TURKISH)
+		language = SCE_SYSTEM_PARAM_LANG_ENGLISH_US; // default to english
+	std::string locale = std::string(vita_locales[language]);
+	sceAppUtilShutdown();
+#elif defined(__3DS__)
+	std::string locale = n3ds::GetLocale();
+#else
+	std::string locale = std::locale("").name().substr(0, 5);
+#endif
+
+	locale = locale.substr(0, 5);
+	LogVerbose("Preferred locale: {}", locale);
+	if (!HasTranslation(locale)) {
+		locale = locale.substr(0, 2);
+		if (!HasTranslation(locale)) {
+			locale = "en";
+		}
+	}
+	LogVerbose("Best match locale: {}", locale);
+
+	GetIniValue(category.data(), key.data(), szCode, sizeof(szCode), locale.c_str());
+}
+void OptionEntryLanguageCode::SaveToIni(string_view category) const
+{
+	SetIniValue(category.data(), key.data(), szCode);
+}
+
+void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
+{
+	if (!languages.empty())
+		return;
+
+	// Add well-known supported languages
+	languages.push_back({ "bg", "Bulgarian" });
+	languages.push_back({ "cs", "Czech" });
+	languages.push_back({ "da", "Danish" });
+	languages.push_back({ "de", "German" });
+	languages.push_back({ "en", "English" });
+	languages.push_back({ "es", "Spanish" });
+	languages.push_back({ "fr", "French" });
+	languages.push_back({ "ja", "Japanese" });
+	languages.push_back({ "hr", "Croatian" });
+	languages.push_back({ "it", "Italian" });
+	languages.push_back({ "ko_KR", "Korean" });
+	languages.push_back({ "pl", "Polish" });
+	languages.push_back({ "pt_BR", "Portuguese" });

```

Comment:
Perhaps Brazilian Portuguese?

Reply:
I don't know the background and how far they differ.
We also use region specific extensions for `ko_KR` and `ro_RO` should we also adjust the text there?
Perhaps @AJenbo or the translator could clarify what they would expect?

Reply:
Brazilian Portuguese is different enough from Portuguese that usually there is a separate translation for it.
Portuguese speakers need to be aware that this is `pt_BR` or else they'll report bugs :) /cc @julealgon 

This is not the case with `ko_KR` or `ro_RO`

Reply:
The Korean governments differ in there policies on language. This coupled with the low cultural exchange rate appears to have caused the languages to start to drift apart. So the `ko_KR` distinction makes sens here. However considering the sanctions, there is little reason to carry out this distinction in the user interface.

The `_RO` seams unnecessary and should be removed for better auto detection.

Reply:
I changed to `Portuguese (Brazil)`. 🙂

Reply:
Good callout by @glebm . They are actually vastly different from one another so the distinction is indeed the way to go.

Sometimes in local apps we have "Portugues" (Portuguese) and "Portugues de Portugal" (Portugal Portuguese) options. That's a more common way we refer to "Portugues" in general over here. However, that's only the case when the information is already presented in pt-BR in the first place. For neutral/english interfaces such as this it makes sense to indicate the country in both options.

A pt-BR translated version of this menu could present "Portugues" (unqualified) and "Portugues de Portugal" as options and it would not be confusing to Brazilian folks.